src/port1.0/portconfigure.tcl

```diff
@@ -340,25 +340,25 @@ proc portconfigure::configure_start {args} {
 
     set compiler [option configure.compiler]
     set valid_compilers {
-        {^apple-gcc-(4\.[02])$}             {MacPorts Apple GCC %s}
-        {^cc$}                              {System cc}
-        {^clang$}                           {Xcode Clang}
-        {^gcc$}                             {System GCC}
-        {^gcc-(3\.3|4\.[02])$}              {Xcode GCC %s}
-        {^llvm-gcc-4\.2$}                   {Xcode LLVM-GCC 4.2}
-        {^macports-clang$}                  {MacPorts Clang (port select)}
-        {^macports-clang-(\d+\.\d+)$}       {MacPorts Clang %s}
-        {^macports-gcc$}                    {MacPorts GCC (port select)}
-        {^macports-gcc-(\d+(?:\.\d+)?)$}    {MacPorts GCC %s}
-        {^macports-llvm-gcc-4\.2$}          {MacPorts LLVM-GCC 4.2}
-        {^macports-g95$}                    {MacPorts G95}
-        {^macports-mpich-default$}          {MacPorts MPICH Wrapper for MacPorts' Default C/C++ Compiler}
-        {^macports-openmpi-default$}        {MacPorts Open MPI Wrapper for MacPorts' Default C/C++ Compiler}
-        {^macports-mpich-clang$}            {MacPorts MPICH Wrapper for Xcode Clang}
-        {^macports-openmpi-clang$}          {MacPorts Open MPI Wrapper for Xcode Clang}
-        {^macports-mpich-clang-(\d+\.\d+)$}
+        {^apple-gcc-(4\.[02])$}                   {MacPorts Apple GCC %s}
+        {^cc$}                                    {System cc}
+        {^clang$}                                 {Xcode Clang}
+        {^gcc$}                                   {System GCC}
+        {^gcc-(3\.3|4\.[02])$}                    {Xcode GCC %s}
+        {^llvm-gcc-4\.2$}                         {Xcode LLVM-GCC 4.2}
+        {^macports-clang$}                        {MacPorts Clang (port select)}
+        {^macports-clang-(\d+\.\d+|[1-9][0-9])$}  {MacPorts Clang %s}
```

Comment:
If I understand the intent correctly, the parenthetical expression would be better written as `(\d+(\.\d+)?)` or perhaps even `([\d.]+)`, as there's very little harm being more flexible than strictly necessary.

Reply:
I considered something like that, but went with the above to enforce the fact that the 'dotless' clang versions start with 10. i.e. `macports-clang-10` is valid, but `macports-clang-9` is not.

Reply:
i.e. I am not sure I agree that being more flexible with the expression is necessarily a good idea here.

Reply:
Well, I don't think that base should be enforcing what the names of the ports can be, beyond the bare minimum that it's "macports-clang" and something identifiable as a version. The downside of allowing more is minimal, since if you write something that matches this regex but doesn't correspond to an existing port you just get a "port not found" error instead of "Invalid value for configure.compiler".

Reply:
OK, I made the change.